.strict-typing

```diff
@@ -341,7 +341,6 @@ homeassistant.components.persistent_notification.*
 homeassistant.components.pi_hole.*
 homeassistant.components.ping.*
 homeassistant.components.plugwise.*
-homeassistant.components.poolsense.*
```

Comment:
Just wondering... (shouldn't block this PR)
Is there an easy way to check if the library has `py.typed` via `script\hassfest\mypy_config.py`?

Reply:
@cdce8p is the person for this question

Reply:
> Is there an easy way to check if the library has `py.typed` via `script\hassfest\mypy_config.py`?

Should be fairly strait forward with `importlib.metadata`. Something like this should work:
```py
from importlib.metadata import files

def has_package_py_typed_file(pkg: str) -> bool:
    return any(path for path in files(pkg) or () if "py.typed" in str(path))
```

The question though is if we should actually remove integrations from that list. The additional mypy checks provide value even without a typed dependency. If the type check is passing already, is it really necessary to remove an integration from that list?

Reply:
> I am now also thinking, can we do it the other way around? Make candidates of integrations that can use strict typing?
> 
> Since there are now discussions about transparency and openness of new dependencies, would it make sense to also add quality requirements to new dependencies?

Not sure if that would be to high of a bar to clear. Yes, adding a `py.typed` file is good but only if you also have type annotations.

Maybe we need to think a bit differently here. What about some kind of bot that creates a score card for dependencies? That would help reviewers and code owners alike as they would know what they could improve.

For example, it could include:
- The license (both the `License-Expression` (if [PEP 639](https://peps.python.org/pep-0639/) gets somewhere) and license files)
- List third party dependency (with their constraints). This could be helpful to prevent something like `aiohttp==3.9.0` in a library.
- Compatibility with (upcoming) Python versions based on the `requires-python`. Especially with poetry it's quite common that libraries (unnecessarily) block future Python releases.
- Run ruff / pylint (with a subset) of rules to detect potential issues like Syntax or DeprecationWarnings
- Detect the use of `eval`
- Do wheels exist?